zabbix_metrics_watcher.py

```diff
@@ -214,14 +217,32 @@ def run(zabbix_host, osbs_master, config, instance):
 
             logger.info("Found build %s in %s", build_name, status)
             build = Build(build_name, cmd_base)
-            if status == 'Pending':
+            if status == 'New':
+                now = datetime.datetime.now()
+                if build_name not in builds_in_new:
+                    builds_in_new[build_name] = now
+                spent_in_new = [(now - start).total_seconds() for start in builds_in_new.values()]
+                max_spent_in_new = max(spent_in_new)
+                if spent_in_new and max_spent_in_new > STUCK_IN_NEW_THRESHOLD_SECS:
+                    logger.info("%s build(s) are in New for, longest: %s sec",
+                                len(builds_in_new), max_spent_in_new)
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", max_spent_in_new)
+
+            elif status == 'Pending':
+                builds_in_new = builds_in_new.pop(build_name, {})
+                if not builds_in_new:
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", 0)
```

Comment:
I don't think 0 is ever true though. Also the `.pop` looks suspicious, and we probably want to being doing post-New procesing for != 'New', not just 'Pending'. How about this?:

```
else:
    spent_in_new = builds_in_new.pop(build_name, None)
    if spent_in_new is not None:
        _send_zabbix_message(..., spent_in_new)

if status == 'Pending':
```

Also I'm not sure about the `STUCK_IN_NEW_THRESHOLD_SECS` thing.

What exactly do we want the Zabbix `new_duration` item to represent?